### ISSUE TITLE
No type X is registered inside GetIt

### ISSUE THREAD

**Issue by bvdwalt** [fluttercommunity/get_it#77]
Hi guys,
I'm only having issues using get_it when running tests in my Flutter app, otherwise it works perfectly. When registering the services I use a mock client, but that's not even the part giving a problem. The test which essentially runs the whole app, _'App loads flight correctly'_, succeeds, the only difference I can tell is that the one is a testWidget test (works), and the failing ones are normal tests. 

Kindly see below errors I am getting, first it says that there is no type registered inside Getit and then when the next test runs and I try to register all the services again it suddenly says that the same type is already registered. I only did the second registering to try and see if it would make any difference. 

My GitHub repo containing the below code as well as the full app can be found [here](https://github.com/bvdwalt/spacex_flights).

Thanks in advance for anyone able to assist.

### Registering code
```
final getIt = GetIt.instance;

void registerServices({bool testing = false}) {
  //getIt.reset();
  getIt.registerLazySingleton(() => FlightProvider());

  if (!testing) {
    getIt.registerLazySingleton<http.Client>(() => Client());
  } else {
    getIt.registerLazySingleton<http.Client>(() => MockClient());
  }
}

class MockClient extends Mock implements http.Client {}
```

## Errors on flutter test

```
PS C:\Users\BennieVanDerWalt\source\spacex_flights> flutter test
00:08 +0 -1: C:\Users\BennieVanDerWalt\source\spacex_flights\test\FlightDataService_test.dart: fetchFlightData returns a flight if the http call completes successfully [E]                                                                  
  'package:get_it/get_it_impl.dart': Failed assertion: line 251 pos 14: 'instanceFactory != null': No type FlightProvider is registered inside GetIt.
   Did you forget to pass an instance name?
  (Did you accidentally do  GetIt sl=GetIt.instance(); instead of GetIt sl=GetIt.instance;did you forget to register it?)
  dart:core                                  _AssertionError._throwNew
  package:get_it/get_it_impl.dart 251:14     _GetItImplementation._findFactoryByNameOrType
  package:get_it/get_it_impl.dart 263:27     _GetItImplementation.get
  ..\lib\src\resources\repository.dart 8:32  new Repository
  FlightDataService_test.dart 15:27          main.<fn>.<fn>

00:08 +0 -2: C:\Users\BennieVanDerWalt\source\spacex_flights\test\FlightDataService_test.dart: fetchFlightData throws an exception if the http call completes with an error [E]
  Invalid argument(s): Type FlightProvider is already registered
  package:get_it/get_it_impl.dart 6:18   throwIf
  package:get_it/get_it_impl.dart 547:5  _GetItImplementation._register
  package:get_it/get_it_impl.dart 403:5  _GetItImplementation.registerLazySingleton
  ..\lib\service_locator.dart 11:9       registerServices
  FlightDataService_test.dart 25:7       main.<fn>.<fn>

00:11 +1 -2: Some tests failed.
```

**Comment by yanivshaked** [comment]
I'm having the same issue - for now, the workaround is to transfer the boolean parameter (test) to the application running inside runApp() and use it there to initialize your services.
(Obviously this is a temporary workaround, waiting to the developer to analyze this)

**Comment by escamoteur** [comment]
I just replaced the import statements from relative to `import 'package:spacex_flights/src/resources/flight_provider.dart';`
Then GetIt works fine but I get another Exception that comes probably from Mocito.

**Comment by bvdwalt** [comment]
Thanks for you input @yanivshaked, but I went for the suggestion by @escamoteur. 

I changed the following import and some others to the package import style:
from:
`import '../models/flights.dart';`
to:
`import 'package:spacex_flights/src/models/flights.dart';`

@escamoteur, I'm not that clued up on Flutter just yet, so is the 'package:' import style the best way to do imports or is this because of some get_it internal workings that it works better that way?

Thanks for your time and looking at my code, my latest commit on my repo has fixed all the tests.
